Makefile

```diff
@@ -1,5 +1,21 @@
 COMMIT = $(shell git rev-parse --short HEAD)$(shell [[ $$(git status --porcelain) = "" ]] || echo -dirty)
 
+ifeq ($(OS),Windows_NT)
+	BUILD_MODE ?= NT
+else
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S), Linux)
+		BUILD_MODE ?= LINUX
+		CMD_SHASUM ?= sha256sum --quiet
+		CMD_FIND ?= find
+	endif
+	ifeq ($(UNAME_S), Darwin)
```

Comment:
I would like to keep build scripts simple and avoid introducing branching based on developer's OS, if possible. In `openshift/openshift-azure` we ask macOS users to setup GNU utils on their machines instead of making changes into build scripts. The `kubernetes/kubernetes` project has the same appraoch.

<details>
  <summary>This is what I do on macOS</summary>
  
```bash
# GNU utils for openshift-azure/kubernetes development
# Ref: https://web.archive.org/web/20190704110904/https://www.topbug.net/blog/2013/04/14/install-and-use-gnu-command-line-tools-in-mac-os-x
# gawk, diffutils, gzip, screen, watch, git, rsync, wdiff
export PATH="/usr/local/bin:$PATH"
# coreutils
export PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
# ed
export PATH="/usr/local/opt/ed/libexec/gnubin:$PATH"
# findutils
export PATH="/usr/local/opt/findutils/libexec/gnubin:$PATH"
# gnu-indent
export PATH="/usr/local/opt/gnu-indent/libexec/gnubin:$PATH"
# gnu-sed
export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
# gnu-tar
export PATH="/usr/local/opt/gnu-tar/libexec/gnubin:$PATH"
# gnu-which
export PATH="/usr/local/opt/gnu-which/libexec/gnubin:$PATH"
# grep
export PATH="/usr/local/opt/grep/libexec/gnubin:$PATH"
# openssh
export PATH="/usr/local/opt/openssl/bin:$PATH"
# binutils
export PATH="/usr/local/opt/binutils/bin:$PATH"
export LDFLAGS="-L/usr/local/opt/binutils/lib"
export CPPFLAGS="-I/usr/local/opt/binutils/include"
# gnu-time
export PATH="/usr/local/opt/gnu-time/libexec/gnubin:$PATH"
```
</details>

This works for v3 plugin development, Kubernetes and for most of the things in v4.

Reply:
Thanks @m1kola for the tips this is very useful and works like a charm now :)

Should we just maybe add this instruction somewhere in the doc ?

Reply:
@julienstroheke yeah, updating docs is a good idea. Could you please a submit a PR for this, if you have a moment?

Reply:
Sure

Reply:
@m1kola, I don't agree with the following because this comment is assuming the steps today is OS agnostic (its not). It is OS specific --> for Linux. 

>>I would like to keep build scripts simple and avoid introducing branching based on developer's OS.

So, I would like us to arrive at a solution such that steps work out of the box for both Linux and Mac OS users (with no specific casing for Mac users to run manual steps that are linked in documentation). This ensure moving forward all development steps will keep both users in mind (we are happy to add Mac OS steps when needed).

Reply:
@amanohar we are open to developers on OSX working on this repo; many on the RH team also use OSX.  Everyone will have some pre-setup to do (install Go, etc.), and there will inevitably be some differences between Linux and OSX pre-setups (`dnf install gpgme-devel` / `brew install gpgme`, etc.) as well.  We are aiming to limit the amount of shell scripting, etc. in the repo, but it seems likely to me that installing the gnu utils on OSX will minimise the chances of unexpected differences in command line flags, usages, etc., and make it easier for everyone to ensure compatibility down the line.

Reply:
There isn't a whole lot of extra shell scripting and avoids extra manual steps on part of OSx developers. This also ensures that both users are kept in mind for tooling.

Reply:
Hi folks @jim-minter, bumping this again.

Reply:
I believe that installing the gnu utils on OSX is a better route technically as it minimises the chances of unexpected differences in command line flags, usages, etc., and makes it easier for everyone to ensure compatibility down the line.  Happy to discuss on a call.  Happy to add documentation.  RH devs are mixed between Linux and OSX and we also want to treat both equally.

Reply:
Sure, happy to discuss on a call.